celery/schedules.py

```diff
@@ -101,6 +101,11 @@ def to_local(self, dt):
             return timezone.to_local_fallback(dt)
         return dt
 
+    def __eq__(self, other):
+        if isinstance(other, BaseSchedule):
+            return other.nowfun == self.nowfun
```

Comment:
I figured an equality check was better in case you pass in a callable (such as a "timezone now" class which stores a timezone attribute and returns the current time with `__call__` based on its stored timezone).  `is` will fail after such an object is pickled and unpickled, even if it evaluates as equal.